test/common/toString.spec.ts

```diff
@@ -47,3 +47,43 @@ describe('global function patch', () => {
        }));
   });
 });
+
+describe('ZoneTask', () => {
+  it('should return handleId.toString if handleId is available', () => {
+    let macroTask1: any = undefined;
+    let macroTask2: any = undefined;
+    let microTask: any = undefined;
+    const zone = Zone.current.fork({
+      name: 'timer',
+      onScheduleTask: (delegate: ZoneDelegate, curr: Zone, target: Zone, task: Task) => {
+        if (task.type === 'macroTask') {
+          if (!macroTask1) {
+            macroTask1 = task;
+          } else {
+            macroTask2 = task;
+          }
+        } else if (task.type === 'microTask') {
+          microTask = task;
+        }
+        return task;
+      }
+    });
+    zone.run(() => {
+      const id1 = setTimeout(() => {});
+      clearTimeout(id1);
+      const id2 = setTimeout(() => {});
+      clearTimeout(id2);
+      Promise.resolve().then(() => {});
+      const macroTask1Str = macroTask1.toString();
+      const macroTask2Str = macroTask2.toString();
+      expect(typeof macroTask1Str).toEqual('string');
+      expect(macroTask1Str).toEqual(id1.toString());
+      expect(typeof macroTask2Str).toEqual('string');
+      expect(macroTask2Str).toEqual(id2.toString());
+      if (macroTask1.data && typeof macroTask1.data.handleId === 'number') {
```

Comment:
Why is this check necessary?

Reply:
Yeah, the reason is this case will also run in `nodejs` environment, in `nodejs` the `handleId` will be a `TimerObject`, in that case, we will not check their equality.